### DISCUSSION TITLE
Can I build a docker image (no push) without push?

### DISCUSSION THREAD

**Discussion by till** [goreleaser/goreleaser#3110] [Q&A] [answered]
Hey 👋🏼  — I am not sure if my Google-fu left me, but I've been banging my head against the wall for a bit.

I added a configuration to build docker images, but I want to test it all before I release because I have to copy a bunch of files into  the image, and I am also trying to figure out if e.g. entrypoint etc. works or if it generally makes sense.

I know I can build a Docker image myself (`docker build ...`), but is there a way to trigger building of the image(s) with `goreleaser` without pushing them, when it's a `--snapshot`? E.g. to test the entire release pipeline before tagging a release?

Thanks,
Till

**Comment by caarlos0** [comment, marked as answer]
Hey till, if you pass either `--snapshot` or `--skip-publish`, nothing is published.

Just make sure to not manually add a `--push` flag to your docker config in your `.goreleaser.yml` :)

**Reply by till** [reply]
Ah, thank you. I keep forgetting about `release`, only my CI runs that.

Is there a reason why `goreleaser build --snapshot --rm-dist` wouldn't run the dockers?

**Reply by caarlos0** [reply]
yes, `goreleaser build` is meant to build binaries only...

**Comment by till** [comment]
Took me a bit — works perfectly, thank again!
